src/python/pants/backend/python/framework/stevedore/target_types.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2023 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+from pants.engine.target import StringSequenceField, Targets
+from pants.util.strutil import softwrap
+
+
+class StevedoreNamespace(str):
+    """Syntactic sugar to tag a namespace in entry_points as a stevedore namespace.
+
+    For example:
+        python_distribution(
+            ...
+            entry_points={
+                stevedore_namespace("a.b.c"): {
+                    "plugin_name": "some.entry:point",
+                },
+            },
+        )
+    """
+
+    alias = "stevedore_namespace"
+
+
+# This is a lot like a SpecialCasedDependencies field, but it doesn't list targets directly.
+class StevedoreNamespacesField(StringSequenceField):
+    alias = "stevedore_namespaces"
+    help = softwrap(
+        """
+        List the stevedore namespaces required by this target.
+
+        Code for all entry_points on python_distribution targets with these
+        namespaces will be added as dependencies so that they are available on
+        PYTHONPATH during tests. Plus, an entry_points.txt file will be generated
+        in the sandbox so that the distribution appears to be "installed".
+        The stevedore namespace format (my.stevedore.extension) is similar
+        to a python namespace.
```

Comment:
The way the plugin is implemented now, you have to define `python_distribution` to define the entry points. But yes, you don't need to duplicate the entry point defs across multiple `python_distributions`. A previous version of this plugin used a new `stevedore_extension` target to define the entry points, but that made getting those entry_points into the `python_distribution`s quite ugly.

I hadn't even considered declaring duplicate `python_distributions` with overlapping sets of entry points. But, I see how that could have been an alternate way of doing things. I still wouldn't want to manually curate a dependencies list of all of the distributions that provide the relevant entry points - it would always be out of date. So, inferring based on the namespace feels more maintainable to me.

I'll think about how I might expand this docstring.

Reply:
Updated. wdyt now?